vec_inf/cli/_cli.py

```diff
@@ -0,0 +1,262 @@
+import os
+
+import click
+import pandas as pd
+from rich.console import Console
+from rich.columns import Columns
+from rich.panel import Panel
+
+from ._utils import *
+
+
+CONSOLE = Console()
+
+
+@click.group()
+def cli():
+    """Vector Inference CLI"""
+    pass
+
+
+@cli.command("launch")
+@click.argument(
+    "model-name",
+    type=str,
+    nargs=1
+)
+@click.option(
+    "--model-family",
+    type=str,
+    help='The model family name according to the directories in `models`'
+)
+@click.option(
+    "--model-variant",
+    type=str,
+    help='The model variant according to the README in `models/model-family`'
+)
+@click.option(
+    "--max-model-len",
+    type=int,
+    help='Model context length. If unspecified, will be automatically derived from the model config.'
+)
+@click.option(
+    "--partition",
+    type=str,
+    help='Type of compute partition, default to a40'
+)
+@click.option(
+    "--num-nodes",
+    type=int,
+    help='Number of nodes to use, default to suggested resource allocation for model'
+)
+@click.option(
+    "--num-gpus",
+    type=int,
+    help='Number of GPUs/node to use, default to suggested resource allocation for model'
+)
+@click.option(
+    "--qos",
+    type=str,
+    help='Quality of service, default to m3'
+)
+@click.option(
+    "--time",
+    type=str,
+    help='Time limit for job, this should comply with QoS, default to 4:00:00'
+)
+@click.option(
+    "--data-type",
+    type=str,
+    help='Model data type, default to auto'
+)
+@click.option(
+    "--venv",
+    type=str,
+    help='Path to virtual environment'
+)
+@click.option(
+    "--log-dir",
+    type=str,
+    help='Path to slurm log directory'
+)
+@click.option(
+    "--json-mode",
+    is_flag=True,
+    help='Output in JSON string',
+)
+def launch(
```

Comment:
Is it still possible for someone to run their own model weights? Assuming it conforms to an architecture supported by vllm? Also, what about passing in additional config for launch? Max token length ect. Is that still possible?

Reply:
If the architecture is supported by vllm, then yes, as long as they supply values for all the optional arguments. W.r.t max tokens, that is set in chat/completions API endpoint, not in model launch.

Reply:
Sorry not max tokens, I meant max_model_len. Can these be set via the launch command?

Reply:
Yes that's an optional argument